dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ManagedExecutorProducer.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.mp.cdi;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorBuilder;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+public class ManagedExecutorProducer {
+
+    private static final TraceComponent tc = Tr.register(ManagedExecutorProducer.class);
+
+    // A new instance of this class will be constructed each time the producer is
+    // called so we need to keep the config->executor map in a static variable
```

Comment:
good catch, I'll add a TODO for now and be sure to add test for this in a follow-up PR.  However, this made me realize that the sharing of identical config across various injection points is probably undesirable in most situations.

If I was writing an app and say I fire async events for payments and notifications, and I happen to configure them the same like this:

```java
public class PaymentProcessor {
    @Inject
    @ManagedExecutorConfig(maxAsync = 5)
    ManagedExecutor paymentExectuor;
}

public class NotificationProcessor {
    @Inject
    @ManagedExecutorConfig(maxAsync = 5)
    ManagedExecutor notificationExecutor;
}
```

I would think that I'm allocating up to 5 threads for payments, and up to 5 threads for notifications.  However, since these ManagedExecutors are the same instance, I would have a max of 5 threads between Payments+Notifications.  Looking at the code, this is not intuitive at all. 
I'd like to discuss this one more, and understand if it's just Liberty's implementation doing this or if the MP Concurrency spec mandates this.

Reply:
Yes, the MP concurrency spec currently mandates it, but the spec is still being developed if the community agrees it is wrong mandated.  This is best to discuss in the spec group.